### PR TITLE
Download and use clojure lsp native binaries

### LINKED ISSUE #1017: Use graalvm-compiled native image for clojure-lsp instead of jar

### PR DESCRIPTION
**Pull request by bpringe**
## What has Changed?

- Calva now downloads the graalvm-compiled native binary for clojure-lsp for the user's OS and uses that instead of using a jar file
- The jar file is no longer packaged with Calva, reducing the extension size
- We no longer have to manually update clojure-lsp, or at least, all that is involved is updating the clojure-lsp version in config.ts
- Clojure-lsp now starts faster and uses less CPU and memory while running
- Adds status indication that clojure-lsp is downloading, then that it's activating
- Download only occurs if the current version of clojure-lsp does not match the configured version, so a download occurs once per machine per clojure-lsp update made in Calva
  - The version is saved to a file after a successful download + unzip
  - Reading the version from a file is _way_ faster than calling the binary to get the version using node. ~0.18 msecs vs ~430 msecs
- Before downloading a new version, the current version of clojure-lsp is backed up. If the download fails, the backup is used

### Todo list
- [x] Test on Mac
- [x] Test on Linux
- [x] Test on Windows
- [ ] Add unit tests 

Closes #1017
Addresses #990

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. NB: *There is a CircleCI bug that makes the Artifacts hard to find. Please see [this issue](https://discuss.circleci.com/t/artifacts-tab-not-showing-unless-logged-in/32433) for workarounds.*
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Added to or updated docs in this branch, if appropriate

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [ ] Made sure the PR is directed at the `dev` branch (unless reasons).
- [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and tested it there if so.
- [ ] Read the source changes.
- [ ] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] If need be, had a chat within the team about particular changes.

Ping @pez, @kstehn, @cfehse, @bpringe

### CODE REVIEW

File: docs/site/clojure-lsp.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ Calva uses a mix of static and dynamic analysis to power the experience. A lot o
 
 ## Starting the LSP server
 
-You don't need to do anything to start clojure-lsp. No install, no commands, no nothing. It does take a while for clojure-lsp to start, though, especially the first time for a new project, when clojure-lsp (via `clj-kondo`) indexes the project files.
+You don't need to do anything to start clojure-lsp. No install, no commands, no nothing. Calva will download the correct binary for your operating system if necessary (this should only happen when the clojure-lsp version is updated in a new release of Calva), and then start it. It does take a while for clojure-lsp to start, though, especially the first time for a new project, when clojure-lsp (via `clj-kondo`) indexes the project files.
```
Comment (PEZ) [inline docs/site/clojure-lsp.md]:
```suggestion
You don't need to do anything to start clojure-lsp. No install, no commands, no nothing. Calva will download the correct binary for your operating system if necessary (this should only happen when the clojure-lsp version is updated in a new release of Calva), and then starts it. It does take a while for clojure-lsp to start, though, especially the first time for a new project, when clojure-lsp (via `clj-kondo`) indexes the project files.
```

**Review by PEZ — APPROVED** [review]
LGTM

File: docs/site/clojure-lsp.md (hunk 1 above)
Comment (bpringe) [reply docs/site/clojure-lsp.md]: Reworded a bit to say "Calva downloads... then starts." I think it did make sense grammatically before as "Calva will download, then [will] start," but now it's maybe a bit more clear.

**Comment by bpringe** [comment]
Just to track here, @PEZ reported an issue with navigating to project source definitions taking a while. He reports the lag does not exist if setting the clojure-lsp version to `2021.02.10-03.01.19` in config.ts, indicating this is likely related to a change in clojure-lsp.

**Comment by HTML-CODDER** [comment]
looks nice

**Comment by bpringe** [comment]
Closing this and switching implementation to TypeScript due to #1074.

**Pull request closed without merging**